action.yml

```diff
@@ -0,0 +1,202 @@
+    required: false
+  pipes-token:
+    description: The Turbot Pipes token used to save snapshots, can also be set via environment variable (https://steampipe.io/docs/reference/env-vars/pipes_token).
+    required: false
+  github-token:
+    description: GitHub token used for git/GitHub operations.
+    required: false
+    default: ${{ github.token }}
+
+runs:
+  using: composite
+  steps:
+    - name: Create temporary directories
+      shell: bash
+      id: make-temp-dir
+      run: |
+        # Create temporary directories for Steampipe and mod installation
+        tmp=$(mktemp -d)
+        tmp_mod_dir=$tmp/mod_dir
+        mkdir -p $tmp_mod_dir
+        echo "tmp-mod-dir=$tmp_mod_dir" >> $GITHUB_OUTPUT
+
+    - name: Get mod
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ inputs.github-token }}
+      run: |
+        # Clone and setup mod
+        if [ -z "${{inputs.mod-branch}}" ]; then
+          git clone --depth 1 "${{ inputs.mod-url }}" "${{ steps.make-temp-dir.outputs.tmp-mod-dir }}"
+        else
+          git clone --depth 1 -b "${{ inputs.mod-branch }}" --single-branch "${{ inputs.mod-url }}" "${{ steps.make-temp-dir.outputs.tmp-mod-dir }}"
+        fi
+        cd ${{ steps.make-temp-dir.outputs.tmp-mod-dir }}
+        steampipe mod install
+        cd -
+
+    - name: Get checks
+      id: get-checks
+      shell: bash
+      run: |
+        # Convert multiline 'checks' input to single line arguments which we can send to 'check'
+        if [ -z "${{ inputs.checks }}" ]
+        then
+          # default to all
+          run_list="all"
+        else
+          run_list=
+          while read line; do
+            run_list="$run_list $line"
+          done <<EOF
+        ${{ inputs.checks }}
+        EOF
+        fi
+        echo "run_list=$run_list" >> $GITHUB_OUTPUT
+
+    - name: Parse exports and artifact paths
+      id: export-paths
+      shell: bash
+      run: |
+        # Parse exports & artifact paths
+        inputs=${{ inputs.artifact-exports }}
+        exports="--export=json --export=md"
+        paths=""
+        date=$(date '+%Y%m%d')
+        if [ "$inputs" != "" ];then
+          for i in ${inputs//,/ }
+          do
+            if [ "$i" != "json" ] && [ "$i" != "md" ];then
+              exports="$exports --export=$i"
+            fi
+            if [[ "$paths" == "" ]];then
+              paths="*$date*.${i}"
+            else
+              paths=$(echo -e "${paths}\n*$date*.${i}")
+            fi
+          done
+        fi
+
+        echo "control_exports=$exports" >> $GITHUB_OUTPUT
+        echo "summary_file_pattern=*$date*.md" >> $GITHUB_OUTPUT
+        echo "artifact_paths<<EOF" >> $GITHUB_OUTPUT
+        echo "$paths" >> $GITHUB_OUTPUT
+        echo "EOF" >> $GITHUB_OUTPUT
+
+    - name: Run checks
+      id: run-checks
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ inputs.github-token }}
```

Comment:
What do we use the GITHUB_TOKEN for in this step?

Reply:
I think this is a hang-over from when I was testing/checking something, will remove.